### PR TITLE
certs.json available through SecurityContext

### PR DESCRIPTION
**Pull request by jaceklaskowski**
`certs.json` is managed by `SecurityContext` and its location is determined by the `UnityCatalogServer`'s configuration (`etc/conf` by default).

Although `etc/conf` configuration directory (folder) can be different in `UnityCatalogServer`, `JwksOperations` used the hardcoded `etc/conf/certs.json` path. There was even a `TODO: This should be configurable`.

This change lets anyone with access to the `SecurityContext` to access `certs.json` whenever it is stored by the UC server.

### CODE REVIEW

**Comment by jaceklaskowski** [comment]
Hey @creechy, mind if I asked you to approve the PR? 🙏  Thank you! 🙇

File: server/src/main/java/io/unitycatalog/server/utils/JwksOperations.java [hunk 1]
```diff
@@ -26,8 +26,10 @@ public class JwksOperations {
 
   private final WebClient webClient = WebClient.builder().build();
   private static final ObjectMapper mapper = new ObjectMapper();
+  private final SecurityContext securityContext;
 
-  public JwksOperations() {
+  public JwksOperations(SecurityContext securityContext) {
```
[the remark this answers is not in the document]
Comment (jaceklaskowski) [reply server/src/main/java/io/unitycatalog/server/utils/JwksOperations.java:31]: It's possible, but `SecurityContext` is as much OK as just a property of it so I don't mind either way.

But, `JwksOperations` is all about security so passing in `SecurityContext` can bring more and encourage to add more to the context if needed. I'm just guessing, but worth considering anyway.

**Comment by jaceklaskowski** [comment]
@josephmock-db @creechy @tdas Mind if I asked you to (re)review and approve this tiny little PR? 🙏

**Comment by jaceklaskowski** [comment]
@dennyglee Can you help me here? 🙏

**Comment by dennyglee** [comment]
Thanks @jaceklaskowski , for this one, I'm inclined to ask @tdas or @creechy for their review.  Thanks!

File: server/src/main/java/io/unitycatalog/server/utils/JwksOperations.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (creechy) [reply server/src/main/java/io/unitycatalog/server/utils/JwksOperations.java:31]: That's fine. It wasn't a strong opinion.

**Review by creechy — APPROVED** [review]
looks good.

**Comment by dennyglee** [comment]
Hey @jaceklaskowski - I noticed that three of the tests are currently failing per the error below.  Any chance you can take a look at this one? 

```
Generated classpath file '/home/runner/work/unitycatalog/unitycatalog/examples/cli/target/classpath'
[error] /home/runner/work/unitycatalog/unitycatalog/server/src/main/java/io/unitycatalog/server/utils/JwksOperations.java:27:1: cannot find symbol
[error]   symbol:   class SecurityContext
[error]   location: class io.unitycatalog.server.utils.JwksOperations
[error] SecurityContext
[error] /home/runner/work/unitycatalog/unitycatalog/server/src/main/java/io/unitycatalog/server/utils/JwksOperations.java:29:1: cannot find symbol
[error]   symbol:   class SecurityContext
[error]   location: class io.unitycatalog.server.utils.JwksOperations
[error] SecurityContext
[error] (server / Compile / compileIncremental) javac returned non-zero exit code
[error] Total time: 26 s, completed Oct 29, 2024, 7:57:41 AM
```

**Comment by jaceklaskowski** [comment]
I'm looking into them, but found no solution yet. No idea why it happens since it works for me on mac os. Still hunting it down...

**Comment by jaceklaskowski** [comment]
@dennyglee At long last! Resolved! Mind if I asked you to merge the changes to `main`? 🙏

**Review by dennyglee — APPROVED** [review]
Thanks a bunch @jaceklaskowski !

**Pull request merged**